### PR TITLE
feat: icon message component

### PR DESCRIPTION
**Pull request by stefanoScalzo**
## Related Issue
Closes SAP/fundamental-styles#1893

## Description
Add the icon message component
the user has the option to just use an icon or just use text

### After:

Screen Shot 2020-11-24 at 4 33 56 PM
Screen Shot 2020-11-24 at 4 33 50 PM
Screen Shot 2020-11-24 at 4 33 45 PM
Screen Shot 2020-11-24 at 4 33 40 PM
Screen Shot 2020-11-24 at 4 33 35 PM

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: stories/icon-message/icon-message.stories.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* You want to show a title using a semantic color
+            `,
+        components: ['icon-message']
+    }
+};
+
+export const success = () => `
+<div>
+<div class="fd-icon-message-success"></div>
+<div class="fd-icon-message-success fd-icon-message-success--no-icon">Success</div>
```
Comment (JKMarkowski) [inline stories/icon-message/icon-message.stories.js:20]: Could you change it to
```suggestion
<i arial-label="success" class="fd-icon-message-success"></i>

<br/><br/>

<span class="fd-icon-message-success fd-icon-message-success--no-icon">Success</span>
```

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
In general - Great Job with `@each` usage, I'm just wondering if it's worth to have `fd-icon-message` component, when this component doesn't have `icon`. I checked the naming
image
Maybe consider changing is somehow to semantic-notification? Or message status?

File: stories/icon-message/icon-message.stories.js (hunk 1 above)
Comment (stefanoScalzo) [reply stories/icon-message/icon-message.stories.js]: i fixed it with different naming convention but same items

File: stories/semantic-color/semantic-color.stories.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+* You want to show a title using a semantic color
+            `,
+        components: ['semantic-color']
+    }
+};
+
+export const information = () => `
+<div>
+    <i arial-label="information" class="fd-semantic-color__icon"></i>
+    <br>
```
Comment (jbadan) [inline stories/semantic-color/semantic-color.stories.js:21]: Can we avoid using `<br />` tags for styling?

**Review by salarenko — CHANGES_REQUESTED** [review]
I don't think this should be structured like this:
image

Doesn't it make more sense to go with:
```
<div class="fd-semantic-color">
    <i class="fd-semantic-color__icon sap-icon--*"></i>
    <span class="fd-semantic-color__text"></span>
</div>
```

Is this really an "Semantic Color" component? Because that's how its named in the Storybook, and here in the PR its "Icon Message".

**Comment by stefanoScalzo** [comment]
> I don't think this should be structured like this:
> image
> 
> Doesn't it make more sense to go with:
> 
> ```
> <div class="fd-semantic-color">
>     <i class="fd-semantic-color__icon sap-icon--*"></i>
>     <span class="fd-semantic-color__text"></span>
> </div>
> ```
> 
> Is this really an "Semantic Color" component? Because that's how its named in the Storybook, and here in the PR its "Icon Message".

Fixed the naming and the ay the documentation is displayed for more clarity

File: stories/semantic-color/semantic-color.stories.js (hunk 2 above)
Comment (stefanoScalzo) [reply stories/semantic-color/semantic-color.stories.js]: Fixed so i dont need line breaks anymore

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
This should not be a stand-alone component. It's part of the icons. Even in the Wiki specs it's shown as part of the icons.

**Comment by stefanoScalzo** [comment]
> This should not be a stand-alone component. It's part of the icons. Even in the Wiki specs it's shown as part of the icons.

I put it inside the icon component like the wiki

**Review by mikerodonnell89 — APPROVED** [review]

**Review by JKMarkowski — APPROVED** [review]
Looks good for me now

File: stories/icon/icon.stories.js [hunk 3]
```diff
@@ -56,3 +56,36 @@ availableIcons.parameters = {
     },
     skipRTLSnapshot: true
 };
+
+
+export const iconSemantics = () => `
+<div>
+    <i arial-label="information" class="fd-icon-message__icon"></i>
```
Comment (JKMarkowski) [inline stories/icon/icon.stories.js:63]: I think it doesn't follow BEM rules

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
There one things that should be changed

**Comment by InnaAtanasova** [comment]
This component is expecting design changes so I close it for now. Thanks for the work @stefanoScalzo

**Pull request closed without merging**